integration-tests/pkg/collector_manager/collector_manager_k8s.go

```diff
@@ -0,0 +1,222 @@
+		{Name: "var-rw", ReadOnly: false, MountPath: "/host/var", MountPropagation: &propagationHostToContainer},
+		{Name: "run-rw", ReadOnly: false, MountPath: "/host/run", MountPropagation: &propagationHostToContainer},
+		{Name: "tmp", ReadOnly: false, MountPath: "/tmp", MountPropagation: &propagationHostToContainer},
+		{Name: "module", ReadOnly: false, MountPath: "/module"},
+	}
+
+	volumes := []coreV1.Volume{
+		{Name: "proc-ro", VolumeSource: coreV1.VolumeSource{HostPath: &coreV1.HostPathVolumeSource{Path: "/proc"}}},
+		{Name: "etc-ro", VolumeSource: coreV1.VolumeSource{HostPath: &coreV1.HostPathVolumeSource{Path: "/etc"}}},
+		{Name: "usr-ro", VolumeSource: coreV1.VolumeSource{HostPath: &coreV1.HostPathVolumeSource{Path: "/usr/lib"}}},
+		{Name: "sys-ro", VolumeSource: coreV1.VolumeSource{HostPath: &coreV1.HostPathVolumeSource{Path: "/sys/kernel/debug"}}},
+		{Name: "var-rw", VolumeSource: coreV1.VolumeSource{HostPath: &coreV1.HostPathVolumeSource{Path: "/var"}}},
+		{Name: "run-rw", VolumeSource: coreV1.VolumeSource{HostPath: &coreV1.HostPathVolumeSource{Path: "/run"}}},
+		{Name: "tmp", VolumeSource: coreV1.VolumeSource{HostPath: &coreV1.HostPathVolumeSource{Path: "/tmp"}}},
+		{Name: "module", VolumeSource: coreV1.VolumeSource{EmptyDir: &coreV1.EmptyDirVolumeSource{}}},
+	}
+
+	return &K8sCollectorManager{
+		executor:     e,
+		volumeMounts: mounts,
+		volumes:      volumes,
+		env:          env,
+		config:       collectorConfig,
+		testName:     name,
+	}
+}
+
+func (k *K8sCollectorManager) Setup(options *CollectorStartupOptions) error {
+	if options == nil {
+		// default values
+		options = &CollectorStartupOptions{}
+	}
+
+	for name, value := range options.Env {
+		k.env = replaceOrAppendEnvVar(k.env, coreV1.EnvVar{Name: name, Value: value})
+	}
+
+	configJson, err := json.Marshal(k.config)
+	if err != nil {
+		return err
+	}
+	k.env = replaceOrAppendEnvVar(k.env, coreV1.EnvVar{Name: "COLLECTOR_CONFIG", Value: string(configJson)})
+
+	if options.Config != nil {
+		maps.Copy(k.config, options.Config)
+	}
+
+	return nil
+}
+
+func (k *K8sCollectorManager) Launch() error {
+	objectMeta := metaV1.ObjectMeta{
+		Name:      "collector",
+		Namespace: TEST_NAMESPACE,
+		Labels:    map[string]string{"app": "collector"},
+	}
+
+	privileged := true
+	container := coreV1.Container{
+		Name:            "collector",
+		Image:           config.Images().CollectorImage(),
+		Ports:           []coreV1.ContainerPort{{ContainerPort: 8080}},
+		Env:             k.env,
+		VolumeMounts:    k.volumeMounts,
+		SecurityContext: &coreV1.SecurityContext{Privileged: &privileged},
+	}
+
+	pod := &coreV1.Pod{
+		ObjectMeta: objectMeta,
+		Spec: coreV1.PodSpec{
+			Containers:    []coreV1.Container{container},
+			Volumes:       k.volumes,
+			RestartPolicy: coreV1.RestartPolicyNever, // if the pod fails, it fails
+		},
+	}
+
+	executor := k.executor.(*executor.K8sExecutor)
+	_, err := executor.CreatePod(TEST_NAMESPACE, pod)
```

Comment:
I believe it will panic either at the moment of the cast or once we try to call `CreatePod` on it, I skipped some checks because we are inside the k8s manager and the only way we can get an instance of it is to specifically use a k8s executor, so it shouldn't be possible for the executor to be of a different type.

That said, if you want me to check on every single cast, that should be fine.